gk/main.c

```diff
@@ -357,7 +378,9 @@ gk_process_request(struct flow_entry *fe, struct ipacket *packet,
 	if (ret < 0)
 		return ret;
 
-	/* TODO Fill up the Ethernet header of the packet. */
+	stale = pkt_copy_cached_eth_header(packet->pkt, eth_cache);
+	if (stale)
```

Comment:
Here and in other places of this patch, we don't need variable `stale`. Rewrite both lines as:
```if (pkt_copy_cached_eth_header(packet->pkt, eth_cache))```